server/src/main/java/io/druid/server/coordination/BatchDataSegmentAnnouncer.java

```diff
@@ -95,18 +103,37 @@ public DataSegment apply(DataSegment input)
         return rv;
       }
     };
+
+    if (this.config.isSkipSegmentAnnouncementOnZk()) {
```

Comment:
Instead of the various changes in thsi class, it might be nice to have a different `DataSegmentAnnouncer` that essentially just adds to the lookups and has `getSegmentChangesSince` on the concrete type rather than the interface.  We then create a `CompositeDataSegmentAnnouncer` which can take both of them and call both.  The endpoint would have only the concrete lookup type injected into it (marked @Nullable and intelligent about returning a `404` if the object is `null`).